### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by dinosaurjr10**
See README.md for details

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,72 @@
...
+* Validates JWTs for both HTTP APIs and server-side web applications
+* Enforces JWT expiration
+* Returns 401 as appropriate for API requests
+* Redirects clients to a configurable log in URL for web applications
+  * Includes an URL to redirect to upon successful authentication
+  * If a GET request, uses the originally requested URL
+  * If a non-GET request, uses HTTP Referrer if available, otherwise uses a configured default URL
+* Implements sliding expiration
+* Supports JWTs in Authorization headers or session cookies
+
```
Comment (peejaybee) [inline README.md]: Might want to put installation instructions in here.

Comment (dinosaurjr10) [inline README.md]: Good point! I'm planning on making a separate PR with more README love.

**Comment by peejaybee** [comment]
This looks good to me.  If Michael and/or the other Phillip like it, it should be merged.

File: lib/jwtap/access_handler.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    end
+  else
+    fail_request var, request, false
+  end
+end
+
+def refresh(var, payload)
+  refreshed_expiration = (Time.now + var.jwtap_expiration_duration_seconds.to_i).to_i
+  refreshed_payload = payload.merge 'exp' => refreshed_expiration
+  JWT.encode refreshed_payload, var.jwtap_secret_key, var.jwtap_algorithm
```
Comment (stlcardsfan) [inline lib/jwtap/access_handler.rb]: Could the self.encode method be used here?

Comment (dinosaurjr10) [inline lib/jwtap/access_handler.rb]: > Could the self.encode method be used here?
@stlcardsfan: Good call, will do.

File: lib/jwtap/access_handler.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+end
+
+def cookie(var)
+  cookies = cookies var
+  cookie = cookies[var.jwtap_cookie_name] if cookies
+  cookie
+end
+
+def cookies(var)
+  Hash[var.http_cookie.split(';').map { |s| s.split('=').map(&:strip) }] if var.http_cookie
```
Comment (stlcardsfan) [inline lib/jwtap/access_handler.rb]: Not sure how likely either issue is:
1.  If a base64 encoding has equals signs at the end then the split on '=' won't get the whole jwt.
2. If any other cookie in the request has an '=' in it then Hash[] can't parse it and will throw an error.

Comment (dinosaurjr10) [inline lib/jwtap/access_handler.rb]: Good catch. Let me revisit this.

Comment (stlcardsfan) [inline lib/jwtap/access_handler.rb]: Cool, the split('=', 2) is a nifty way to solve it.

**Comment by stlcardsfan** [comment]
Looks good

**Pull request merged**